crates/fluvio-sc/src/services/public_api/create.rs

```diff
@@ -24,13 +24,23 @@ pub async fn handle_create_request<AC: AuthContext, C: MetadataItem>(
     let (header, req) = request.get_header_request();
 
     debug!(?req, "create request");
-    let status = if let Some(create) = req.downcast()? as Option<CreateRequest<TopicSpec>> {
-        super::topic::handle_create_topics_request(create, auth_context).await?
-    } else if let Some(create) = req.downcast()? as Option<CreateRequest<SpuGroupSpec>> {
+
+    let status = if let Some(create) = req.downcast()? as Option<CreateRequest<SpuGroupSpec>> {
         super::spg::handle_create_spu_group_request(create, auth_context).await?
     } else if let Some(create) = req.downcast()? as Option<CreateRequest<CustomSpuSpec>> {
         super::spu::RegisterCustomSpu::handle_register_custom_spu_request(create, auth_context)
             .await
+    } else if auth_context.global_ctx.config().read_only_metadata {
```

Comment:
There is already `auth_ctx`, which checks permission.  Instead of adding another case, could create read-only auth_ctx so there is no need to create different path

Reply:
updated